### PR TITLE
Add ProcessRunner so running processes is testable.

### PR DESCRIPTION
**Pull request by hterkelsen**
Also add some tests for the `test` command.

### CODE REVIEW

File: test/util.dart [hunk 1]
```diff
@@ -16,42 +18,55 @@ void initializeFakePackages() {
 
 /// Creates a plugin package with the given [name] under the mock packages
 /// directory.
-void createFakePlugin(
+Directory createFakePlugin(
   String name, {
   bool withSingleExample: false,
   List<String> withExamples: const <String>[],
+  bool withTestDir: false,
```
Comment (collinjackson) [inline test/util.dart]: One thing that I did differently in my implementation that I provided a way to add arbitrary extra (empty) files.

      createFakePlugin('plugin', withExtraFiles: [
        <String>['example', 'test', 'plugin_e2e.dart']
        <String>['example', 'test_driver', 'plugin_e2e_test.dart']
        <String>['example', 'android', 'gradlew'],
        <String>['example', 'android', 'app', 'src', 'androidTest', 'MainActivityTest.java'],
      ];

Comment (hterkelsen) [reply test/util.dart]: I like that, much better than adding a bool flag for every possible configuration. I'll update my PR

Comment (hterkelsen) [reply test/util.dart]: Done.

**Comment by hterkelsen** [comment]
Thanks. Added `withExtraFiles`, PTAL

**Pull request merged**